### PR TITLE
Add initial table

### PR DESCRIPTION
**Pull request by bigdatasourav**

### CODE REVIEW

File: docs/tables/nomad_acl_binding_rule.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  a.name as auth_method_name,
+  a.type as auth_method_type,
+  a.create_time as auth_method_creation_time,
+  b.id as binding_rule_id,
+  b.bind_type as bind_type
+from
+  nomad_acl_binding_rule as b
+  left join
+    nomad_acl_auth_method as a
+    on b.auth_method = a.name;
```
Comment (madhushreeray30) [inline docs/tables/nomad_acl_binding_rule.md]:
```suggestion
  left join nomad_acl_auth_method as a on b.auth_method = a.name;
```

File: docs/tables/nomad_acl_token.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  global,
+  create_time,
+  expiration_ttl
+from
+  nomad_acl_token
+where
+  global;
+```
+
+### List tokens which will not expire
```
Comment (madhushreeray30) [inline docs/tables/nomad_acl_token.md]:
```suggestion
### List tokens which will never expire
```

File: docs/tables/nomad_job.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# Table: nomad_job
+
+A Job is a specification provided by users that declares a workload for Nomad. A Job is a form of desired state; the user is expressing that the job should be running, but not where it should be run. The responsibility of Nomad is to make sure the actual state matches the user desired state. A Job is composed of one or more task groups.
```
Comment (madhushreeray30) [inline docs/tables/nomad_job.md]:
```suggestion
A job is a specification provided by users that declares a workload for Nomad. A job is a form of desired state; the user is expressing that the job should be running, but not where it should be run. The responsibility of Nomad is to make sure the actual state matches the user desired state. A job is composed of one or more task groups.
```

File: docs/tables/nomad_deployment.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# Table: nomad_deployment
+
+Deployments are the mechanism by which Nomad rolls out changes to cluster state in a step-by-step fashion. Deployments are only available for Jobs with the type service. When an Evaluation is processed, the scheduler creates only the number of Allocations permitted by the update block and the current state of the cluster. The Deployment is used to monitor the health of those Allocations and emit a new Evaluation for the next step of the update.
```
Comment (madhushreeray30) [inline docs/tables/nomad_deployment.md]:
```suggestion
Deployments are the mechanism by which Nomad rolls out changes to cluster state in a step-by-step fashion. Deployments are only available for Jobs with the type service. When an Evaluation is processed, the scheduler creates only the number of Allocations permitted by the update block and the current state of the cluster. The deployment is used to monitor the health of those Allocations and emit a new Evaluation for the next step of the update.
```

File: docs/tables/nomad_node.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  node_class,
+  drain,
+  status,
+  datacenter,
+  cgroup_parent
+from
+  nomad_node;
+```
+
+### List nodes with drain set to true
```
Comment (madhushreeray30) [inline docs/tables/nomad_node.md]:
```suggestion
### List nodes with drain enabled
```

File: docs/tables/nomad_plugin.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  id,
+  title,
+  version,
+  create_index,
+  modify_index,
+  provider
+from
+  nomad_plugin
+where
+  controller_rquired;
```
Comment (madhushreeray30) [inline docs/tables/nomad_plugin.md]:
```suggestion
  controller_required;
```

File: docs/tables/nomad_volume.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  clone_id,
+  capacity,
+  namespace,
+  provider,
+  plugin_id
+from
+  nomad_volume;
+```
+
+### List volumes with the access mode set to `ReadWriteOnce`
```
Comment (madhushreeray30) [inline docs/tables/nomad_volume.md]:
```suggestion
### List volumes with access mode set to `ReadWriteOnce`
```

File: README.md [hunk 8]
```diff
@@ -1 +1,137 @@
...
+
+Use SQL to query nodes, jobs, deployments and more from Nomad.
+
+- **[Get started →](https://hub.steampipe.io/plugins/turbot/nomad)**
+- Documentation: [Table definitions & examples](https://hub.steampipe.io/plugins/turbot/nomad/tables)
+- Community: [Slack Channel](https://steampipe.io/community/join)
+- Get involved: [Issues](https://github.com/turbot/steampipe-plugin-nomad/issues)
+
+## Quick start
+
```
Comment (madhushreeray30) [inline README.md:13]:
```suggestion

### Install

```

File: README.md [hunk 9]
```diff
@@ -1 +1,137 @@
...
+Download and install the latest Nomad plugin:
+
+```bash
+steampipe plugin install nomad
+```
+
+Configure your [credentials](https://hub.steampipe.io/plugins/turbot/nomad#credentials) and [config file](https://hub.steampipe.io/plugins/turbot/nomad#configuration).
+
+### Configuring Nomad Credentials
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
```

File: README.md [hunk 10]
```diff
@@ -1 +1,137 @@
...
+
+- `address`: The address of the nomad server.
+- `namespace`: The Nomad Cluster namespace.
+
+```hcl
+connection "nomad" {
+  plugin    = "nomad"
+  address   = "http://18.118.144.168:4646"
+  namespace = "*"
+}
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
connection "nomad" {
  plugin    = "nomad"
  
  # Authentication information
  address   = "http://18.118.144.168:4646"
  namespace = "*"
}
```

File: README.md [hunk 11]
```diff
@@ -1 +1,137 @@
...
+- `address`: The address of the nomad server.
+- `namespace`: The Nomad Cluster namespace.
+- `secret_id`: The SecretID of an ACL token.
+
+```hcl
+connection "nomad" {
+  plugin    = "nomad"
+  address   = "http://18.118.144.168:4646"
+  namespace = "*"
+  secret_id = "c178b810-8b18-6f38-016f-725ddec5d58"
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
connection "nomad" {
  plugin    = "nomad"
  
  # Authentication information
  address   = "http://18.118.144.168:4646"
  namespace = "*"
  secret_id = "c178b810-8b18-6f38-016f-725ddec5d58"
```

File: README.md [hunk 12]
```diff
@@ -1 +1,137 @@
...
+
+```hcl
+connection "nomad" {
+  plugin    = "nomad"
+  address   = "http://18.118.144.168:4646"
+  namespace = "*"
+}
+```
+
+or you may specify the Address, Namespace and SecretID to authenticate:
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
Or you may specify the Address, Namespace and SecretID to authenticate:
```

File: README.md [hunk 13]
```diff
@@ -1 +1,137 @@
...
+```hcl
+connection "nomad" {
+  plugin    = "nomad"
+  address   = "http://18.118.144.168:4646"
+  namespace = "*"
+  secret_id = "c178b810-8b18-6f38-016f-725ddec5d58"
+}
+```
+
+or through environment variables
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
Or through environment variables:
```

File: README.md [hunk 14]
```diff
@@ -1 +1,137 @@
...
+
+```sh
+export NOMAD_ADDR="http://18.118.144.168:4646"
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
export NOMAD_ADDR=http://18.118.144.168:4646
```

File: README.md [hunk 15]
```diff
@@ -1 +1,137 @@
...
+export NOMAD_NAMESPACE="*"
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
export NOMAD_NAMESPACE=*
```

File: README.md [hunk 16]
```diff
@@ -1 +1,137 @@
...
+export NOMAD_TOKEN="c178b810-8b18-6f38-016f-725ddec5d58"
```
Comment (madhushreeray30) [inline README.md]:
```suggestion
export NOMAD_TOKEN=c178b810-8b18-6f38-016f-725ddec5d58
```

File: docs/index.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+
+```sh
+steampipe plugin install nomad
+```
+
+### Credentials
+
+| Item        | Description                                                                                                                                                                              |
+| ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Credentials | Nomad requires an Address and Namespace or Address, Namespace and [Secret ID](https://developer.hashicorp.com/nomad/tutorials/access-control/access-control-tokens) for all requests.    |
```
Comment (madhushreeray30) [inline docs/index.md]:
```suggestion
| Credentials | Nomad requires an `address` and `namespace` or `address`, `namespace` and [Secret ID](https://developer.hashicorp.com/nomad/tutorials/access-control/access-control-tokens) for all requests.    |
```

File: config/nomad.spc [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+  # For more information on the Namespace, please see https://developer.hashicorp.com/nomad/tutorials/manage-clusters/namespaces.
+  # Can also be set with the NOMAD_NAMESPACE environment variable.
+  # "*" indicates all the namespaces available.
+  namespace = "*"
+
+  # `secret_id` - The SecretID of an ACL token.
+  # The SecretID is required to make requests for ACL-enabled clusters.
+  # For more information on the ACL Token, please see https://developer.hashicorp.com/nomad/tutorials/access-control/access-control-tokens.
+  # Can also be set with the NOMAD_TOKEN environment variable.
+  # secret_id = "c178b810-8b18-6f38-016f-725ddec5d58"
```
Comment (madhushreeray30) [inline config/nomad.spc:11]: Should this be secret_id or token? @misraved

File: docs/index.md [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+
+```hcl
+connection "nomad" {
+  plugin    = "nomad"
+  address   = "http://18.118.144.168:4646"
+  namespace = "*"
+}
+```
+
+or you may specify the Address, Namespace and SecretID to authenticate:
```
Comment (madhushreeray30) [inline docs/index.md]:
```suggestion
Or you may specify the Address, Namespace and SecretID to authenticate:
```

File: docs/index.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+  plugin    = "nomad"
+  address   = "http://18.118.144.168:4646"
+  namespace = "*"
+  secret_id = "c178b810-8b18-6f38-016f-725ddec5d58"
+}
+```
+
+or through environment variables
+
+The Nomad plugin will use the Nomad environment variable to obtain credentials **only if the `address`,`namespace` and `secret_id` is not specified** in the connection:
```
Comment (madhushreeray30) [inline docs/index.md]:
```suggestion
Alternatively, you can also use the standard Namecheap environment variables to obtain credentials **only if the `address`,`namespace` and `secret_id` is not specified** in the connection:
```

File: docs/index.md [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+
+```sh
+export NOMAD_ADDR="http://18.118.144.168:4646"
```
Comment (madhushreeray30) [inline docs/index.md]:
```suggestion
export NOMAD_ADDR=http://18.118.144.168:4646
```

File: docs/index.md [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+export NOMAD_NAMESPACE="*"
```
Comment (madhushreeray30) [inline docs/index.md]:
```suggestion
export NOMAD_NAMESPACE=*
```

File: docs/index.md [hunk 23]
```diff
@@ -0,0 +1,136 @@
...
+export NOMAD_TOKEN="c178b810-8b18-6f38-016f-725ddec5d58"
```
Comment (madhushreeray30) [inline docs/index.md]:
```suggestion
export NOMAD_TOKEN=c178b810-8b18-6f38-016f-725ddec5d58
```

File: go.mod [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+require (
+	github.com/gorilla/websocket v1.5.0 // indirect
+	github.com/hashicorp/cronexpr v1.1.1 // indirect
+	github.com/hashicorp/errwrap v1.0.0 // indirect
+	github.com/hashicorp/go-cleanhttp v0.5.2 // indirect
+	github.com/hashicorp/go-multierror v1.1.1 // indirect
+	github.com/hashicorp/go-rootcerts v1.0.2 // indirect
+	github.com/mitchellh/go-homedir v1.1.0 // indirect
+	github.com/mitchellh/mapstructure v1.5.0 // indirect
+	github.com/turbot/steampipe-plugin-sdk/v5 v5.3.0
```
Comment (madhushreeray30) [inline go.mod]:
```suggestion
	github.com/turbot/steampipe-plugin-sdk/v5 v5.4.1
```

Comment (madhushreeray30) [reply go.mod]: Run go mod tidy

File: nomad/table_nomad_acl_binding_rule.go [hunk 25]
```diff
@@ -0,0 +1,162 @@
...
+			},
+			{
+				Name:        "description",
+				Type:        proto.ColumnType_STRING,
+				Description: "The description of the acl binding rule.",
+			},
+			{
+				Name:        "auth_method",
+				Type:        proto.ColumnType_STRING,
+				Description: "The name of the auth method for which this rule applies to.",
```
Comment (madhushreeray30) [inline nomad/table_nomad_acl_binding_rule.go]:
```suggestion
				Description: "The name of the auth method for which this rule applies.",
```

**Review by misraved — APPROVED** [review]

**Pull request merged**